src/main/java/pl/javastart/task/CustomList.java

```diff
@@ -1,6 +1,76 @@
 package pl.javastart.task;
 
+import java.util.Arrays;
+
 // Uzupełnij klasę. Możesz ją dowolnie modyfikować.
 // Celem jest, żeby przechodziły testy w src/test/java/[...]/CustomListTest
-public class CustomList {
-}
+public class CustomList<T> {
+
+    private Object[] customs;
+    private int size;
+
+    public CustomList() {
+        customs = new Object[10];
+        size = 0;
+    }
+
+    public T get(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
+        return (T) customs[index];
+    }
+
+    public void add(T element) {
+        if (size == customs.length) {
+            increaseCapacity();
+        }
+        customs[size++] = element;
+    }
+
+    public void add(int index, T element) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
+        if (size == customs.length) {
+            increaseCapacity();
+        }
+        for (int i = size - 1; i >= index; i--) {
+            customs[i + 1] = customs[i];
+        }
```

Comment:
Zamiast takiego ręcznego kopiowania użyj System.arrayCopy, żeby znacząco to przyspieszyć